### ISSUE TITLE
100_ == ParserError but 100__ is ok.

### ISSUE THREAD

**Issue by unc0** [ckknight/gorillascript#5]
I'm making gorillascript vim syntax highlight support. When I was testing these numbers, I found this problem, maybe it's a bug?

**Comment by ckknight** [comment]
Hmm, due to the way numbers are parsed, `1_0_0_0 == 1000` and `1_comment == 1`, so what's happening is the second underscore is seen as the comment. I could make it so that numbers ending in a trailing underscore are allowed, I suppose.

**Comment by ckknight** [comment]
Alright, fixed in 44b33629eae88c4df33fe055532e32b4aa8094e2.

The following is valid (which wasn't before):

```
// trailing underscore
1_ == 1
// underscores surrounding the decimal point
1_2_._3_4_ == 12.34
```
